### ISSUE TITLE
Question aboutSGD

### ISSUE THREAD

**Issue by zengwbz** [ifzhang/FairMOT#229]
Hello When I just replace ADam with SGD：
optimizer = torch.optim.SGD([{'params': model.parameters(), 'weight_decay': 5e-4}],
                                lr=opt.lr, momentum=0.9,nesterov=True)
The Loss goes down very quickly, but id_loss Stay in around 8.5095
|loss 2.3018 |hm_loss 0.7770 |wh_loss 2.7907 |off_loss 0.2367 |id_loss 8.5095 |Data 0.001s(0.001s) |Net 0.308s

**Comment by ifzhang** [comment]
You may check whether you add the parameters of the id_classifier to the optimizer:
https://github.com/ifzhang/FairMOT/blob/2a353b16a591bf997aadd5d24b73e599672906c4/src/lib/trains/base_trainer.py#L30
